tensorflow/cc/gradients/math_grad.cc

```diff
@@ -728,6 +728,26 @@ Status LgammaGrad(const Scope& scope, const Operation& op,
 }
 REGISTER_GRADIENT_OP("Lgamma", LgammaGrad);
 
+Status SelectGrad(const Scope& scope, const Operation& op,
+                  const std::vector<Output>& grad_inputs,
+                  std::vector<Output>* grad_outputs) {
+  auto comparator = op.input(0);
+  auto x = op.input(1);
+  auto zeros = ZerosLike(scope, x);
+  auto grad = grad_inputs[0];
+
+  Scope grad_scope = scope.WithControlDependencies(grad);
```

Comment:
Why do you need this control dependency? Isn't there a data dependency with grad already?